### ISSUE TITLE
Tests do not pass. Error code 2. "Misuse of shell builtins". STDERR

### ISSUE THREAD

**Issue by miguelgf** [infection/infection#61]
| Question    | Answer
| ------------| ---------------
| Infection version | 0.6.0 (`infection.phar --version`)
| Test Framework version | PHPUnit 6.2.4
| PHP version | 7.1.4
| Platform    | MacOS
| Github Repo | Private repo

I've just installed and try to use the infection command, and I get "Misuse of shell builtins" error.

```
> infection --threads=4
Running initial test suite...

Phpunit version: 6.2.4

 2871 [============================] 2 minsTests do not pass. Error code 2. "Misuse of shell builtins". STDERR: 
````
<details>
 <summary>phpunit.xml</summary>
 
 ```xml
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/6.2/phpunit.xsd"
         backupGlobals="false"
         colors="true"
         bootstrap="autoload.php"
         beStrictAboutTestsThatDoNotTestAnything="true"
         beStrictAboutOutputDuringTests="true"
         beStrictAboutTestSize="true"
         beStrictAboutChangesToGlobalState="true"
>
    <testsuites>
        <testsuite name="Project Test Suite">
            <directory>../src/*/*Bundle/Tests</directory>
            <directory>../src/*/Bundle/*Bundle/Tests</directory>
            <directory>../src/*Bundle/Tests</directory>
            <directory>../src/Application/Tests</directory>
            <directory>../src/Domain/Tests</directory>
        </testsuite>
    </testsuites>

    <php>
        <ini name="memory_limit" value="512m"/>
    </php>

    <filter>
        <whitelist>
            <directory>../src</directory>
            <exclude>
                <directory>../src/*Bundle/Resources</directory>
                <directory>../src/*Bundle/Tests</directory>
                <directory>../src/*/*Bundle/Resources</directory>
                <directory>../src/*/*Bundle/Tests</directory>
                <directory>../src/*/Bundle/*Bundle/Resources</directory>
                <directory>../src/*/Bundle/*Bundle/Tests</directory>
            </exclude>
        </whitelist>
    </filter>

    <listeners>
        <listener class="AppBundle\Test\Listener\IntegrationTestsFixturesLoaderListener" />
    </listeners>
</phpunit>

 ```
</details>

**Comment by borNfreee** [comment]
Hello,

it's hard to say without the code what is actually triggers such error. It can be some method that requires (s)tty or anything else that does not work correctly under symfony process.

Since you have a private repo, I can't help you. Just a suggestion - could you please localize the "bad" test and try to understand what is the reason?

**Comment by miguelgf** [comment]
Is there a way to to this? Can I activate some verbose or debug output?

**Comment by borNfreee** [comment]
Instead of running all test, you can pass an option to run only, for example, one folder with tests. If they still fail, you can run a particular test from that folder and so one until you find the "bad" test.

```bash
infection.phar --test-framework-options="--verbose --filter=some/folder"
```

Or you can try to [disable](https://askubuntu.com/questions/357039/how-do-i-disable-virtual-consoles-tty1-6) tty on your terminal (depends on OS) and try to run unit tests for your project without infection.

**Comment by borNfreee** [comment]
I've added test framework output to the console, could you please try Infection on `dev-master`. Probably, it will help a lot with determining what test fails.

**Comment by miguelgf** [comment]
I will try this on monday and let you know. Thanks.

**Comment by miguelgf** [comment]
With this debug info I could locate the test that was failing, on the test suite it was also included the symfony's integration tests, and when executing the tests from another location it didnt work without `KERNEL_DIR` constant on phpunit.xml ([doc](https://symfony.com/doc/current/testing.html#your-first-functional-test)).

I think you should keep this log output, it's very useful.

**Comment by borNfreee** [comment]
Yeah, it is already in master and will be released in 0.6.1.

Do you think absence of `KERNEL_DIR` is bug in Infection? If so, could you please create an issue for that?

**Comment by miguelgf** [comment]
> Do you think absence of KERNEL_DIR is bug in Infection? If so, could you please create an issue for that?

Not at all, it's our application problem, we didn't realized because we always execute the phpunit from the same location.

Can I close this one?

Yes, thanks for the help and support.
